packages/client/src/client.d.ts

```diff
@@ -26,12 +29,49 @@ declare class Client {
   /**
    * Create request
    */
-  createRequest(data: ClientRequest): ClientRequest;
+  createRequest<TData>(data: Client.RequestOptions<TData>): ClientRequest;
 
   /**
    * Do a request
    */
-  request(data: ClientRequest, cb?: (err: ResponseError, response: [ClientResponse, any]) => void): Promise<[ClientResponse, any]>;
+  request<TData>(opts: Client.RequestOptions<TData>, cb?: (err: ResponseError, response: [ClientResponse, any]) => void): Promise<[ClientResponse, any]>;
+}
+
+declare namespace Client {
+  export interface RequestOptions<TData = any, TParams = object> {
```

Comment:
This looks like mostly a duplicate of `Request.RequestOptions`.

Reply:
So I've played around with this a bit, and haven't been able to import Request.RequestOptions. We have a similar duplication in `twilio-node`: [request.d.ts](https://github.com/twilio/twilio-node/blob/master/lib/http/request.d.ts) vs. [RequestClient.d.ts](https://github.com/twilio/twilio-node/blob/master/lib/base/RequestClient.d.ts)

Reply:
Figured out a workaround